nengo/spa/vocab.py

```diff
@@ -81,6 +81,10 @@ def __init__(self, dimensions, randomize=True, unitary=False,
         self.include_pairs = include_pairs
         self._identity = None
         self.rng = rng
+        self.read_only = False
+
+        # The parent of a non-subset vocabulary is itself
+        self.parent = self
```

Comment:
Hmm... this seems like a red flag to me that this implementation is overly complex. Do you really need parent references? It makes managing memory and ensuring that these object get garbage collected very complicated...